src/functions-and-arrays.js

```diff
@@ -1,41 +1,112 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(number1, number2) {
+
+  if (number1 === number2) {
+    return number1
+  } else if (number2 > number1) {
+    return number2
+  } else {
+    return number1
+  }
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  let longest
+
+  if (words.length === 0) {
+    return null
+  }
+  for (let i = 0; i < words.length; i++) {
+    const word = words[i]
+
+    if (longest === undefined) {
+      longest = word 
+    } else if (word.length > longest.length){
+      longest = word
+    }
+  }
+  return longest
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0
+
+  for (let i = 0; i < numbers.length; i++) {
+    const number = numbers[i]
+
+    if (sum === undefined) {
+      sum = number
+    } else {
+      sum += number
+    }
+  }
+  return sum
+}
```

Comment:
En el fondo sum nunca va a ser undefined porque siempre lo estas iniciando a 0, entonces por aqui no entra yo te firia de quitar el if, y para ahorrarnos espacio en la memoria podemos sumar directamente el numero sin necesidad de la constante (pero si te queda mas claro con las constante se deja que no pasa nada!!)

```suggestion
function sumNumbers(numbers) {
  let sum = 0
  for (let i = 0; i < numbers.length; i++) {
    sum += numbers[i]
  }
  return sum
}
```

Reply:
Es verdad! el undefined aqui no tiene sentido.